diff --git a/CMakeLists.txt b/CMakeLists.txt
index 521071800..cb58c611b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -395,12 +395,12 @@ option(LLVM_INCLUDE_EXAMPLES "Generate build targets for the LLVM examples" ON)
 
 option(LLVM_BUILD_TESTS
   "Build LLVM unit tests. If OFF, just generate build targets." OFF)
-option(LLVM_INCLUDE_TESTS "Generate build targets for the LLVM unit tests." ON)
+option(LLVM_INCLUDE_TESTS "Generate build targets for the LLVM unit tests." OFF)
 
-option(HLSL_INCLUDE_TESTS  "Generate build targets for the HLSL unit tests." ON)
+option(HLSL_INCLUDE_TESTS  "Generate build targets for the HLSL unit tests." OFF)
 
 option (LLVM_BUILD_DOCS "Build the llvm documentation." OFF)
-option (LLVM_INCLUDE_DOCS "Generate build targets for llvm documentation." ON)
+option (LLVM_INCLUDE_DOCS "Generate build targets for llvm documentation." OFF)
 option (LLVM_ENABLE_DOXYGEN "Use doxygen to generate llvm API documentation." OFF)
 option (LLVM_ENABLE_SPHINX "Use Sphinx to generate llvm documentation." OFF)
 
@@ -624,11 +624,6 @@ include_directories( ${LLVM_INCLUDE_DIR} ${LLVM_MAIN_INCLUDE_DIR})
 
 include_directories( ${LLVM_INCLUDE_DIR}/dxc/Tracing) # HLSL Change
 
-# when crosscompiling import the executable targets from a file
-if(LLVM_USE_HOST_TOOLS)
-  include(CrossCompile)
-endif(LLVM_USE_HOST_TOOLS)
-
 if(${CMAKE_SYSTEM_NAME} MATCHES "(FreeBSD|DragonFly)")
   # On FreeBSD, /usr/local/* is not used by default. In order to build LLVM
   # with libxml2, iconv.h, etc., we must add /usr/local paths.
diff --git a/cmake/modules/TableGen.cmake b/cmake/modules/TableGen.cmake
index 85d720e91..579061e0c 100644
--- a/cmake/modules/TableGen.cmake
+++ b/cmake/modules/TableGen.cmake
@@ -92,25 +92,6 @@ macro(add_tablegen target project)
   set(${project}_TABLEGEN_EXE ${${project}_TABLEGEN} PARENT_SCOPE)
   set(${project}_TABLEGEN_TARGET ${${project}_TABLEGEN} PARENT_SCOPE)
 
-  if(LLVM_USE_HOST_TOOLS)
-    if( ${${project}_TABLEGEN} STREQUAL "${target}" )
-      if (NOT CMAKE_CONFIGURATION_TYPES)
-        set(${project}_TABLEGEN_EXE "${LLVM_NATIVE_BUILD}/bin/${target}")
-      else()
-        set(${project}_TABLEGEN_EXE "${LLVM_NATIVE_BUILD}/Release/bin/${target}")
-      endif()
-      set(${project}_TABLEGEN_EXE ${${project}_TABLEGEN_EXE} PARENT_SCOPE)
-
-      add_custom_command(OUTPUT ${${project}_TABLEGEN_EXE}
-        COMMAND ${CMAKE_COMMAND} --build . --target ${target} --config Release
-        DEPENDS CONFIGURE_LLVM_NATIVE ${target}
-        WORKING_DIRECTORY ${LLVM_NATIVE_BUILD}
-        COMMENT "Building native TableGen...")
-      add_custom_target(${project}-tablegen-host DEPENDS ${${project}_TABLEGEN_EXE})
-      set(${project}_TABLEGEN_TARGET ${project}-tablegen-host PARENT_SCOPE)
-    endif()
-  endif()
-
   if( MINGW )
     if(CMAKE_SIZEOF_VOID_P MATCHES "8")
       set_target_properties(${target} PROPERTIES LINK_FLAGS -Wl,--stack,16777216)
diff --git a/include/dxc/Support/dxcapi.use.h b/include/dxc/Support/dxcapi.use.h
index e1aa7aff8..a86e2ab37 100644
--- a/include/dxc/Support/dxcapi.use.h
+++ b/include/dxc/Support/dxcapi.use.h
@@ -16,8 +16,8 @@
 
 namespace dxc {
 
-extern const char *kDxCompilerLib;
-extern const char *kDxilLib;
+extern "C" HRESULT __stdcall DxcCreateInstance(REFCLSID rclsid, REFIID riid, _Out_ LPVOID *ppv);
+extern "C" HRESULT __stdcall DxcCreateInstance2(IMalloc *pMalloc, REFCLSID rclsid, REFIID riid, _Out_ LPVOID *ppv);
 
 // Interface for common dll operations
 class DllLoader {
@@ -60,128 +60,35 @@ public:
 // Helper class to dynamically load the dxcompiler or a compatible libraries.
 class SpecificDllLoader : public DllLoader {
 
-  HMODULE m_dll;
-  DxcCreateInstanceProc m_createFn;
-  DxcCreateInstance2Proc m_createFn2;
-
-  HRESULT InitializeInternal(LPCSTR dllName, LPCSTR fnName) {
-    if (m_dll != nullptr)
-      return S_OK;
-
-#ifdef _WIN32
-    m_dll = LoadLibraryA(
-        dllName); // CodeQL [SM01925] This is by design, intended to be used to
-                  // test multiple validators versions.
-    if (m_dll == nullptr)
-      return HRESULT_FROM_WIN32(GetLastError());
-    m_createFn = (DxcCreateInstanceProc)GetProcAddress(m_dll, fnName);
-
-    if (m_createFn == nullptr) {
-      HRESULT hr = HRESULT_FROM_WIN32(GetLastError());
-      FreeLibrary(m_dll);
-      m_dll = nullptr;
-      return hr;
-    }
-#else
-    m_dll = ::dlopen(
-        dllName, RTLD_LAZY); // CodeQL [SM01925] This is by design, intended to
-                             // be used to test multiple validators versions.
-    if (m_dll == nullptr)
-      return E_FAIL;
-    m_createFn = (DxcCreateInstanceProc)::dlsym(m_dll, fnName);
-
-    if (m_createFn == nullptr) {
-      ::dlclose(m_dll);
-      m_dll = nullptr;
-      return E_FAIL;
-    }
-#endif
-
-    // Only basic functions used to avoid requiring additional headers.
-    m_createFn2 = nullptr;
-    char fnName2[128];
-    size_t s = strlen(fnName);
-    if (s < sizeof(fnName2) - 2) {
-      memcpy(fnName2, fnName, s);
-      fnName2[s] = '2';
-      fnName2[s + 1] = '\0';
-#ifdef _WIN32
-      m_createFn2 = (DxcCreateInstance2Proc)GetProcAddress(m_dll, fnName2);
-#else
-      m_createFn2 = (DxcCreateInstance2Proc)::dlsym(m_dll, fnName2);
-#endif
-    }
-
-    return S_OK;
-  }
-
 public:
-  SpecificDllLoader()
-      : m_dll(nullptr), m_createFn(nullptr), m_createFn2(nullptr) {}
-
+  SpecificDllLoader() {}
   ~SpecificDllLoader() override { Cleanup(); }
 
   HRESULT InitializeForDll(LPCSTR dll, LPCSTR entryPoint) {
-    return InitializeInternal(dll, entryPoint);
+    return S_OK;
   }
 
   // Also bring visibility into the interface definition of this function
   // which takes 2 args
   HRESULT CreateInstanceImpl(REFCLSID clsid, REFIID riid,
                              IUnknown **pResult) override {
-    if (pResult == nullptr)
-      return E_POINTER;
-    if (m_dll == nullptr)
-      return E_FAIL;
-    HRESULT hr = m_createFn(clsid, riid, (LPVOID *)pResult);
-    return hr;
+    return DxcCreateInstance(clsid, riid, (LPVOID *)pResult);
   }
 
   // Also bring visibility into the interface definition of this function
   // which takes 3 args
   HRESULT CreateInstance2Impl(IMalloc *pMalloc, REFCLSID clsid, REFIID riid,
                               IUnknown **pResult) override {
-    if (pResult == nullptr)
-      return E_POINTER;
-    if (m_dll == nullptr)
-      return E_FAIL;
-    if (m_createFn2 == nullptr)
-      return E_FAIL;
-    HRESULT hr = m_createFn2(pMalloc, clsid, riid, (LPVOID *)pResult);
-    return hr;
+    return DxcCreateInstance2(pMalloc, clsid, riid, (LPVOID *)pResult);
   }
 
-  bool HasCreateWithMalloc() const { return m_createFn2 != nullptr; }
+  bool HasCreateWithMalloc() const { return true; }
 
-  bool IsEnabled() const override { return m_dll != nullptr; }
-
-  bool GetCreateInstanceProcs(DxcCreateInstanceProc *pCreateFn,
-                              DxcCreateInstance2Proc *pCreateFn2) const {
-    if (pCreateFn == nullptr || pCreateFn2 == nullptr || m_createFn == nullptr)
-      return false;
-    *pCreateFn = m_createFn;
-    *pCreateFn2 = m_createFn2;
-    return true;
-  }
+  bool IsEnabled() const override { return true; }
 
   void Cleanup() {
-    if (m_dll != nullptr) {
-      m_createFn = nullptr;
-      m_createFn2 = nullptr;
-#ifdef _WIN32
-      FreeLibrary(m_dll);
-#else
-      ::dlclose(m_dll);
-#endif
-      m_dll = nullptr;
-    }
   }
 
-  HMODULE Detach() {
-    HMODULE hModule = m_dll;
-    m_dll = nullptr;
-    return hModule;
-  }
 };
 
 // This class is for instances where we *only* expect
@@ -189,7 +96,7 @@ public:
 class DxCompilerDllLoader : public SpecificDllLoader {
 public:
   HRESULT Initialize() {
-    return InitializeForDll(kDxCompilerLib, "DxcCreateInstance");
+    return S_OK;
   }
 };
 
@@ -200,7 +107,7 @@ public:
 class DXCLibraryDllLoader : public SpecificDllLoader {
 public:
   HRESULT Initialize() {
-    return InitializeForDll(kDxCompilerLib, "DxcCreateInstance");
+    return S_OK;
   }
 };
 
diff --git a/include/dxc/dxcapi.h b/include/dxc/dxcapi.h
index 95cc56a04..07fe01d08 100644
--- a/include/dxc/dxcapi.h
+++ b/include/dxc/dxcapi.h
@@ -15,7 +15,7 @@
 
 #ifdef _WIN32
 #ifndef DXC_API_IMPORT
-#define DXC_API_IMPORT __declspec(dllimport)
+#define DXC_API_IMPORT
 #endif
 #else
 #ifndef DXC_API_IMPORT
diff --git a/lib/DxcSupport/dxcapi.extval.cpp b/lib/DxcSupport/dxcapi.extval.cpp
index b50fd05e0..eea5a68d0 100644
--- a/lib/DxcSupport/dxcapi.extval.cpp
+++ b/lib/DxcSupport/dxcapi.extval.cpp
@@ -466,7 +466,7 @@ DxcDllExtValidationLoader::InitializeForDll(LPCSTR dllName, LPCSTR fnName) {
 }
 
 HRESULT DxcDllExtValidationLoader::initialize() {
-  return InitializeForDll(kDxCompilerLib, "DxcCreateInstance");
+  return E_FAIL;
 }
 
 } // namespace dxc
diff --git a/lib/DxcSupport/dxcapi.use.cpp b/lib/DxcSupport/dxcapi.use.cpp
index fed47e2e8..74727947d 100644
--- a/lib/DxcSupport/dxcapi.use.cpp
+++ b/lib/DxcSupport/dxcapi.use.cpp
@@ -21,11 +21,6 @@
 
 namespace dxc {
 
-const char *kDxCompilerLib =
-    CMAKE_SHARED_LIBRARY_PREFIX "dxcompiler" CMAKE_SHARED_LIBRARY_SUFFIX;
-const char *kDxilLib =
-    CMAKE_SHARED_LIBRARY_PREFIX "dxil" CMAKE_SHARED_LIBRARY_SUFFIX;
-
 #ifdef _WIN32
 static void TrimEOL(char *pMsg) {
   char *pEnd = pMsg + strlen(pMsg);
diff --git a/tools/clang/tools/dxcompiler/CMakeLists.txt b/tools/clang/tools/dxcompiler/CMakeLists.txt
index 26bf0e5d9..cf6ae254c 100644
--- a/tools/clang/tools/dxcompiler/CMakeLists.txt
+++ b/tools/clang/tools/dxcompiler/CMakeLists.txt
@@ -54,8 +54,6 @@ set(SOURCES
   dxcompilerobj.cpp
   dxcvalidator.cpp
   DXCompiler.cpp
-  DXCompiler.rc
-  DXCompiler.def
   dxcfilesystem.cpp
   dxcutil.cpp
   dxcdisassembler.cpp
@@ -124,7 +122,7 @@ if (MSVC)
   find_package(DiaSDK REQUIRED) # Used for constants and declarations.
 endif (MSVC)
 
-add_clang_library(dxcompiler SHARED ${SOURCES})
+add_clang_library(dxcompiler STATIC ${SOURCES})
 add_dependencies(dxcompiler TablegenHLSLOptions) 
 if (MSVC)
   # No DxcEtw on non-Windows platforms.
@@ -151,12 +149,6 @@ set_target_properties(dxcompiler
   VERSION ${LIBCLANG_LIBRARY_VERSION}
   DEFINE_SYMBOL _CINDEX_LIB_)
 
-if (WIN32)
-  set(install_dest RUNTIME DESTINATION bin)
-else()
-  set(install_dest LIBRARY DESTINATION lib${LLVM_LIBDIR_SUFFIX})
-endif()
-
 install(TARGETS dxcompiler
   ${install_dest}
   COMPONENT dxcompiler)
diff --git a/tools/clang/tools/dxcompiler/DXCompiler.cpp b/tools/clang/tools/dxcompiler/DXCompiler.cpp
index c7ffcbffa..9ae51c89f 100644
--- a/tools/clang/tools/dxcompiler/DXCompiler.cpp
+++ b/tools/clang/tools/dxcompiler/DXCompiler.cpp
@@ -94,15 +94,16 @@ void __attribute__((destructor)) DllShutdown() {
   DxcCleanupThreadMalloc();
 }
 #else  // LLVM_ON_UNIX
-BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD Reason, LPVOID reserved) {
-  BOOL result = TRUE;
-  if (Reason == DLL_PROCESS_ATTACH) {
+extern "C" HRESULT DXC_Init(void)
+{
     EventRegisterMicrosoft_Windows_DXCompiler_API();
     DxcEtw_DXCompilerInitialization_Start();
     HRESULT hr = InitMaybeFail();
     DxcEtw_DXCompilerInitialization_Stop(hr);
-    result = SUCCEEDED(hr) ? TRUE : FALSE;
-  } else if (Reason == DLL_PROCESS_DETACH) {
+    return hr;
+}
+extern "C" void DXC_Done(void)
+{
     DxcEtw_DXCompilerShutdown_Start();
     DxcSetThreadMallocToDefault();
     ::hlsl::options::cleanupHlslOptTable();
@@ -112,8 +113,5 @@ BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD Reason, LPVOID reserved) {
     DxcCleanupThreadMalloc();
     DxcEtw_DXCompilerShutdown_Stop(S_OK);
     EventUnregisterMicrosoft_Windows_DXCompiler_API();
-  }
-
-  return result;
 }
 #endif // LLVM_ON_UNIX
diff --git a/tools/clang/tools/dxcompiler/dxcapi.cpp b/tools/clang/tools/dxcompiler/dxcapi.cpp
index d4e85bc35..2a96cbdf0 100644
--- a/tools/clang/tools/dxcompiler/dxcapi.cpp
+++ b/tools/clang/tools/dxcompiler/dxcapi.cpp
@@ -12,7 +12,7 @@
 #include "dxc/Support/WinIncludes.h"
 
 #ifdef _WIN32
-#define DXC_API_IMPORT __declspec(dllexport)
+#define DXC_API_IMPORT
 #else
 #define DXC_API_IMPORT __attribute__((visibility("default")))
 #endif
diff --git a/tools/llvm-config/CMakeLists.txt b/tools/llvm-config/CMakeLists.txt
index edbd8c950..c7ee5f107 100644
--- a/tools/llvm-config/CMakeLists.txt
+++ b/tools/llvm-config/CMakeLists.txt
@@ -47,7 +47,6 @@ if(CMAKE_CROSSCOMPILING)
     WORKING_DIRECTORY ${LLVM_NATIVE_BUILD}
     COMMENT "Building native llvm-config...")
   add_custom_target(${project}NativeLLVMConfig DEPENDS ${${project}_LLVM_CONFIG_EXE})
-  add_dependencies(${project}NativeLLVMConfig CONFIGURE_LLVM_NATIVE)
 
   add_dependencies(llvm-config ${project}NativeLLVMConfig)
 endif(CMAKE_CROSSCOMPILING)
