diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d049f4..47020a1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,7 +121,6 @@ if(SDLSHADERCROSS_VENDORED)
 		add_subdirectory(external/DirectXShaderCompiler EXCLUDE_FROM_ALL)
 		add_library(DirectXShaderCompiler::dxcompiler ALIAS dxcompiler)
 		add_library(DirectXShaderCompiler::dxil ALIAS dxildll)
-		list(APPEND vendored_targets dxcompiler dxildll)
 		add_dependencies(dxcompiler dxildll)
 	endif()
 
diff --git a/src/SDL_shadercross.c b/src/SDL_shadercross.c
index 0e69c4e..a585976 100644
--- a/src/SDL_shadercross.c
+++ b/src/SDL_shadercross.c
@@ -310,6 +310,8 @@ struct IDxcUtils
 #if defined(SDL_PLATFORM_XBOXONE) || defined(SDL_PLATFORM_XBOXSERIES)
 extern HRESULT __stdcall DxcCreateInstance(REFCLSID rclsid, REFIID riid, LPVOID* ppv);
 #else
+HRESULT DXC_Init(void);
+void DXC_Done(void);
 extern HRESULT DxcCreateInstance(REFCLSID rclsid, REFIID riid, LPVOID *ppv);
 #endif
 
@@ -2288,11 +2290,21 @@ bool SDL_ShaderCross_Init(void)
         }
     }
 
+#if !defined(SDL_PLATFORM_XBOXONE) && !defined(SDL_PLATFORM_XBOXSERIES)
+    if (DXC_Init() < 0) {
+        return false;
+    }
+#endif
+
     return true;
 }
 
 void SDL_ShaderCross_Quit(void)
 {
+#if !defined(SDL_PLATFORM_XBOXONE) && !defined(SDL_PLATFORM_XBOXSERIES)
+    DXC_Done();
+#endif
+
     if (d3dcompiler_dll != NULL) {
         SDL_UnloadObject(d3dcompiler_dll);
         d3dcompiler_dll = NULL;
