diff --git a/cmake/PkgConfig.cmake b/cmake/PkgConfig.cmake
index ec4f5383..c9e52a54 100644
--- a/cmake/PkgConfig.cmake
+++ b/cmake/PkgConfig.cmake
@@ -29,7 +29,7 @@ function(set_libs_private out_var)
     foreach(lib IN LISTS ARGN)
         get_filename_component(name "${lib}" NAME)
         foreach(prefix IN LISTS CMAKE_FIND_LIBRARY_PREFIXES)
-            if(name MATCHES "^${prefix}")
+            if("${prefix}" AND name MATCHES "^${prefix}")
                 string(REGEX REPLACE "^${prefix}" "" name "${name}")
                 break()
             endif()
